static/app/views/settings/organizationAuthTokens/authTokenRow.spec.tsx

```diff
@@ -0,0 +1,159 @@
+        token: {
+          ...token,
+          dateLastUsed: new Date(),
+        },
+      };
+
+      render(<OrganizationAuthTokensAuthTokenRow {...props} />);
+
+      expect(await screen.findByLabelText('Token preview')).toHaveTextContent(
+        'sntrys_************XYZ1'
+      );
+      expect(
+        screen.getByText(
+          textWithMarkupMatcher('a few seconds ago in project Project Name')
+        )
+      ).toBeInTheDocument();
+      expect(screen.getByText('My Token')).toBeInTheDocument();
+    });
+
+    it('shows last used project only', async function () {
+      const props = {
+        ...defaultProps,
+        projectLastUsed: TestStubs.Project(),
+        token: {
+          ...token,
+        },
+      };
+
+      render(<OrganizationAuthTokensAuthTokenRow {...props} />);
+
+      expect(screen.getByLabelText('Token preview')).toHaveTextContent(
+        'sntrys_************XYZ1'
+      );
+      expect(
+        await screen.findByText(textWithMarkupMatcher('in project Project Name'))
+      ).toBeInTheDocument();
+      expect(screen.getByText('My Token')).toBeInTheDocument();
+    });
+
+    it('shows last used date only', async function () {
+      const props = {
+        ...defaultProps,
+        token: {
+          ...token,
+          dateLastUsed: new Date(),
+        },
+      };
+
+      render(<OrganizationAuthTokensAuthTokenRow {...props} />);
+
+      expect(await screen.findByLabelText('Token preview')).toHaveTextContent(
+        'sntrys_************XYZ1'
+      );
+      expect(
+        screen.getByText(textWithMarkupMatcher('a few seconds ago'))
+      ).toBeInTheDocument();
+      expect(screen.getByText('My Token')).toBeInTheDocument();
+    });
+  });
+
+  describe('revoking', function () {
+    it('does not allow to revoke without access', async function () {
+      const props = {
+        ...defaultProps,
+        revokeToken: undefined,
+      };
+
+      render(<OrganizationAuthTokensAuthTokenRow {...props} />);
+
+      expect(await screen.findByRole('button', {name: 'Revoke My Token'})).toBeDisabled();
+    });
+
+    it('allows to revoke', async function () {
+      render(<OrganizationAuthTokensAuthTokenRow {...defaultProps} />);
+      renderGlobalModal();
+
+      expect(await screen.findByRole('button', {name: 'Revoke My Token'})).toBeEnabled();
+
+      await userEvent.click(screen.getByRole('button', {name: 'Revoke My Token'}));
+      // Confirm modal
+      await userEvent.click(screen.getByRole('button', {name: 'Confirm'}));
+
+      expect(revokeToken).toHaveBeenCalledWith(token);
+    });
+
+    it('does not allow to revoke while revoking in progress', async function () {
+      const props = {
+        ...defaultProps,
+        isRevoking: true,
+      };
+
+      render(<OrganizationAuthTokensAuthTokenRow {...props} />);
+
+      expect(await screen.findByRole('button', {name: 'Revoke My Token'})).toBeDisabled();
+    });
+  });
+});
```

Comment:
hmm, true! I more or less copied this from the index.spec.tsx, but there I think it is different because the data loading actually happens in the component... so you're totally right!